### ISSUE TITLE
[Bug]: When running official examples and customer asynchronous code with Ascend-vLLM using the external_launcher, both cases experience process blocking issues.

### ISSUE THREAD

**Issue by zhenjie521** [vllm-project/vllm-ascend#1496]
### Your current environment

<details>
 official examples：
import gc

import torch
import torch_npu
import vllm_ascend
from vllm import LLM, SamplingParams
from vllm.distributed.parallel_state import (destroy_distributed_environment,
                                             destroy_model_parallel)

def clean_up():
    destroy_model_parallel()
    destroy_distributed_environment()
    gc.collect()
    torch.npu.empty_cache()

prompts = [
    "Hello, my name is",
    "The future of AI is",
]
sampling_params = SamplingParams(temperature=0.6, top_p=0.95, top_k=40)
llm = LLM(model="/home/c84334770/weights/qwen2.5-math-1.5b/",
          tensor_parallel_size=4,
          distributed_executor_backend="external_launcher",
          seed=1,
          max_model_len=4096)

outputs = llm.generate(prompts, sampling_params)
for output in outputs:
    prompt = output.prompt
    generated_text = output.outputs[0].text
    print(f"Prompt: {prompt!r}, Generated text: {generated_text!r}")

del llm
clean_up()

customer asynchronous code：
import gc
import uuid
import torch
import asyncio
import torch_npu

from vllm import LLM, SamplingParams
from vllm.distributed.parallel_state import (destroy_distributed_environment,
                                             destroy_model_parallel)
import vllm
from vllm.engine.arg_utils import AsyncEngineArgs
from vllm.v1.engine.async_llm import AsyncLLM
import asyncio
# from vllm import AsyncEngineArgs, AsyncLLM, SamplingParams
# from vllm import SamplingParams,AsyncEngineArgs,AsyncLLM
import os
os.environ["VLLM_USE_V1"] = "1"
async def main():
    engine_args = AsyncEngineArgs(
    model="/home/c84334770/weights/qwen2.5-math-1.5b/",
    tensor_parallel_size=4,
    pipeline_parallel_size=1,
    max_model_len=4096,
    dtype="bfloat16",
    gpu_memory_utilization=0.7,
    trust_remote_code=True,
    enforce_eager=False,
    seed=1,
    distributed_executor_backend="external_launcher")
    llm = AsyncLLM.from_engine_args(engine_args)
    prompts ="Hello, my name is"
    # 异步生成
    async_gen = llm.generate(
        prompts,
        sampling_params=SamplingParams(max_tokens=100),
        request_id="test_123"
    )
    
    
    # 必须用 async for 迭代
    async for output in async_gen:
        print(f"生成结果: {output.outputs[0].text}")
        # sleep()

# 运行异步函数
asyncio.run(main())

Runtime environment：
torch=2.5.1
torch_npu=2.5.1
CANN=8.1.rc1
vllm=0.8.5post1
vllm_ascend=0.8.5rc1
</details>

### 🐛 Describe the bug

The situation encountered is a process deadlock during driver initialization.We are given a problem where the system works when tensor parallelism (TP) is 1, but encounters a deadlock during driver initialization when TP > 1.
INFO 06-28 14:26:35 [importing.py:17] Triton not installed or not compatible; certain GPU-related functions will not be available.
WARNING 06-28 14:26:35 [importing.py:28] Triton is not installed. Using dummy decorators. Install it via `pip install triton` to enable kernelcompilation.
INFO 06-28 14:26:35 [importing.py:53] Triton module has been replaced with a placeholder.
INFO 06-28 14:26:37 [__init__.py:30] Available plugins for group vllm.platform_plugins:
INFO 06-28 14:26:37 [__init__.py:32] name=ascend, value=vllm_ascend:register
INFO 06-28 14:26:37 [__init__.py:34] all available plugins for group vllm.platform_plugins will be loaded.
INFO 06-28 14:26:37 [__init__.py:36] set environment variable VLLM_PLUGINS to control which plugins to load.
INFO 06-28 14:26:37 [__init__.py:44] plugin ascend loaded.
INFO 06-28 14:26:37 [__init__.py:230] Platform plugin ascend is activated
WARNING 06-28 14:26:39 [_custom_ops.py:21] Failed to import from vllm._C with ModuleNotFoundError("No module named 'vllm._C'")
INFO 06-28 14:26:41 [__init__.py:30] Available plugins for group vllm.general_plugins:
INFO 06-28 14:26:41 [__init__.py:32] name=ascend_enhanced_model, value=vllm_ascend:register_model
INFO 06-28 14:26:41 [__init__.py:34] all available plugins for group vllm.general_plugins will be loaded.
INFO 06-28 14:26:41 [__init__.py:36] set environment variable VLLM_PLUGINS to control which plugins to load.
INFO 06-28 14:26:41 [__init__.py:44] plugin ascend_enhanced_model loaded.
WARNING 06-28 14:26:41 [registry.py:389] Model architecture DeepSeekMTPModel is already registered, and will be overwritten by the new model class vllm_ascend.models.deepseek_mtp:CustomDeepSeekMTP.
WARNING 06-28 14:26:41 [registry.py:389] Model architecture Qwen2VLForConditionalGeneration is already registered, and will be overwritten by the new model class vllm_ascend.models.qwen2_vl:AscendQwen2VLForConditionalGeneration.
WARNING 06-28 14:26:41 [registry.py:389] Model architecture Qwen2_5_VLForConditionalGeneration is already registered, and will be overwritten by the new model class vllm_ascend.models.qwen2_5_vl:AscendQwen2_5_VLForConditionalGeneration.
WARNING 06-28 14:26:41 [registry.py:389] Model architecture DeepseekV2ForCausalLM is already registered, and will be overwritten by the new model class vllm_ascend.models.deepseek_v2:CustomDeepseekV2ForCausalLM.
WARNING 06-28 14:26:41 [registry.py:389] Model architecture DeepseekV3ForCausalLM is already registered, and will be overwritten by the new model class vllm_ascend.models.deepseek_v2:CustomDeepseekV3ForCausalLM.
INFO 06-28 14:26:54 [config.py:717] This model supports multiple tasks: {'generate', 'embed', 'classify', 'reward', 'score'}. Defaulting to 'generate'.
INFO 06-28 14:26:54 [arg_utils.py:1669] npu is experimental on VLLM_USE_V1=1. Falling back to V0 Engine.
INFO 06-28 14:26:54 [config.py:1729] Disabling V1 multiprocessing for external launcher.
INFO 06-28 14:26:54 [config.py:1804] Disabled the custom all-reduce kernel because it is not supported on current platform.
WARNING 06-28 14:26:54 [platform.py:126] NPU compilation support pending. Will be available in future CANN and torch_npu releases. NPU graph mode is currently experimental and disabled by default. You can just adopt additional_config={'enable_graph_mode': True} to serve deepseek models with NPU graph mode on vllm-ascend with V0 engine. 
INFO 06-28 14:26:54 [platform.py:134] Compilation disabled, using eager mode by default
INFO 06-28 14:26:54 [llm_engine.py:240] Initializing a V0 LLM engine (v0.8.5.post1) with config: model='/home/c84334770/weights/qwen2.5-math-1.5b/', speculative_config=None, tokenizer='/home/c84334770/weights/qwen2.5-math-1.5b/', skip_tokenizer_init=False, tokenizer_mode=auto, revision=None, override_neuron_config=None, tokenizer_revision=None, trust_remote_code=False, dtype=torch.bfloat16, max_seq_len=4096, download_dir=None, load_format=auto, tensor_parallel_size=4, pipeline_parallel_size=1, disable_custom_all_reduce=True, quantization=None, enforce_eager=False, kv_cache_dtype=auto,  device_config=npu, decoding_config=DecodingConfig(guided_decoding_backend='auto', reasoning_backend=None), observability_config=ObservabilityConfig(show_hidden_metrics=False, otlp_traces_endpoint=None, collect_model_forward_time=False, collect_model_execute_time=False), seed=1, served_model_name=/home/c84334770/weights/qwen2.5-math-1.5b/, num_scheduler_steps=1, multi_step_stream_outputs=True, enable_prefix_caching=None, chunked_prefill_enabled=False, use_async_output_proc=True, disable_mm_preprocessor_cache=False, mm_processor_kwargs=None, pooler_config=None, compilation_config={"level":0,"splitting_ops":[],"compile_sizes":[],"cudagraph_capture_sizes":[256,248,240,232,224,216,208,200,192,184,176,168,160,152,144,136,128,120,112,104,96,88,80,72,64,56,48,40,32,24,16,8,4,2,1],"max_capture_size":256}, use_cached_outputs=False, 
/root/miniconda3/envs/gcore/lib/python3.10/site-packages/torch_npu/dynamo/torchair/__init__.py:8: UserWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html. The pkg_resources package is slated for removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools<81.
  import pkg_resources
WARNING 06-28 14:26:55 [utils.py:2522] Methods add_prompt_adapter,cache_config,compilation_config,current_platform,list_prompt_adapters,load_config,pin_prompt_adapter,remove_prompt_adapter not implemented in <vllm_ascend.worker.worker.NPUWorker object at 0xfffd2b2a6200>
>>>>>>>>>>>>>>>device<<<<<<<<<<<<<<<<<<<<<<< npu:0
